android/2023-emmsale/app/src/main/java/com/emmsale/presentation/ui/eventSearch/EventSearchViewModel.kt

```diff
@@ -0,0 +1,91 @@
+package com.emmsale.presentation.ui.eventSearch
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.asLiveData
+import androidx.lifecycle.viewModelScope
+import com.emmsale.data.common.retrofit.callAdapter.ApiResponse
+import com.emmsale.data.common.retrofit.callAdapter.Failure
+import com.emmsale.data.common.retrofit.callAdapter.NetworkError
+import com.emmsale.data.common.retrofit.callAdapter.Success
+import com.emmsale.data.common.retrofit.callAdapter.Unexpected
+import com.emmsale.data.model.Event
+import com.emmsale.data.model.EventSearch
+import com.emmsale.data.repository.interfaces.EventRepository
+import com.emmsale.data.repository.interfaces.EventSearchRepository
+import com.emmsale.presentation.common.FetchResult
+import com.emmsale.presentation.ui.eventSearch.uistate.EventSearchUiState
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.debounce
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.filter
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.mapLatest
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class EventSearchViewModel @Inject constructor(
+    private val eventRepository: EventRepository,
+    private val eventSearchRepository: EventSearchRepository,
+) : ViewModel() {
+    val eventSearchQuery = MutableStateFlow(INITIAL_QUERY)
+    val eventSearchResults: LiveData<EventSearchUiState> = eventSearchQuery
+        .debounce(SEARCH_DEBOUNCE_TIME)
+        .filter { query -> query.trim().isNotEmpty() }
```

Comment:
```suggestion
        .filter { query -> query.isNotBlank() }
```
동일한 동작을 하는 걸로 알고 있습니다 :)

Reply:
반영 완료하였습니다 : )